### PR TITLE
NEAR: Finalization events receiving and IBC messages submission

### PR DESCRIPTION
**Pull request by vmarkushin**

### CODE REVIEW

File: near/src/provider.rs [hunk 1]
```diff
@@ -240,47 +244,43 @@ impl IbcProvider for Client {
...
-	}
-
-	fn remove_packets(&mut self, _seqs: Vec<Sequence>) {
+	async fn consensus_height(&self, _client_height: Height) -> Option<Height> {
 		unimplemented!()
 	}
 
-	fn cached_packets(&self) -> &Vec<Packet> {
-		unimplemented!()
+	async fn latest_height(&self) -> Result<Height, Self::Error> {
```
Comment (seunlanlege) [inline near/src/provider.rs:251]: This method is actually for checking the latest height of the on-chain near light client.

**Review by seunlanlege — CHANGES_REQUESTED** [review]

File: near/src/provider.rs (hunk 1 above)
Comment (Wizdave97) [reply near/src/provider.rs:251]: Light client latest height is gotten by fetching the client state, we use this method  for fetching the latest height of the parachain in parachain implementation.

File: near/src/provider.rs [hunk 2]
```diff
@@ -240,47 +244,43 @@ impl IbcProvider for Client {
 		self.send_query(query).await
 	}
 
-	fn cache_send_packet_seq(&mut self, _packet: Packet) {
-		unimplemented!()
-	}
-
-	fn remove_packets(&mut self, _seqs: Vec<Sequence>) {
+	async fn consensus_height(&self, _client_height: Height) -> Option<Height> {
```
Comment (Wizdave97) [inline near/src/provider.rs:247]: I think you can remove this function everywhere, it was there when we had caching implemented.

File: src/messages.rs [hunk 3]
```diff
@@ -440,8 +440,6 @@ impl Messages {
 					let value = msg.encode_vec();
 					let msg = Any { value, type_url: msg.type_url() };
 					messages.push(msg);
-					// Cache send packets, so we can track timeouts later on
```
Comment (Wizdave97) [inline src/messages.rs:443]: Probably rebase your branch.

File: near/src/provider.rs (hunk 1 above)
Comment (vmarkushin) [reply near/src/provider.rs:251]: But why do we want to have the light client on the near side itself?

File: src/messages.rs (hunk 3 above)
Comment (vmarkushin) [reply src/messages.rs:443]: There was a compilation error, I think the lines leaked here after my previous rebase...

File: near/src/provider.rs (hunk 1 above)
Comment (seunlanlege) [reply near/src/provider.rs:251]: oh sorry about that then, forgot my own apis 😛

Comment (vmarkushin) [reply near/src/provider.rs:251]: it happens :)

File: near/src/lib.rs [hunk 4]
```diff
@@ -41,8 +41,8 @@ pub struct Client {
 }
 
 impl Client {
-	pub fn new(contract_id: AccountId, secret_key: near_crypto::SecretKey) -> Self {
-		openssl_probe::init_ssl_cert_env_vars();
+	/// Initializes configurations for indexer with a default config.
+	pub fn init_configs_default() -> anyhow::Result<()> {
```
Comment (seunlanlege) [inline near/src/lib.rs:45]: this is what i meant by config struct:

https://github.com/ComposableFi/hyperspace/blob/778b705d3450b94e0f5e48de3e878a6b4b10c071/parachain/src/lib.rs#L69-L96

It'll be useful when we introduce a cli interface that pulls config from toml files

Comment (vmarkushin) [reply near/src/lib.rs:45]: Got you

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**